Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/typeselection/SrsTypeSelectionFragment.kt

```diff
@@ -1,17 +1,120 @@
+                                viewModel.srsTestType.value ?: SrsSubmissionType.SRS_OTHER
+                            )
+                    )
+            }
+        }
+
+        val backCallback = object : OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() = viewModel.onCancel()
+        }
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner, backCallback)
+
+        binding.toolbar.setNavigationOnClickListener { viewModel.onCancel() }
+
+        viewModel.srsTestType.observe2(this) {
+            updateButtons(it)
+        }
+    }
+
+    private fun updateButtons(submissionType: SrsSubmissionType?) {
+
+        binding.targetButtonRatRegisteredNoResult.apply {
+            handleColors(submissionType, SrsSubmissionType.SRS_RAT)
+            setOnClickListener { viewModel.onRatRegisteredNoResult() }
+        }
+
+        binding.targetButtonRatNotRegistered.apply {
+            // TODO: RAT unregistered
+            handleColors(submissionType, SrsSubmissionType.SRS_RAT)
+            setOnClickListener { viewModel.onRatNotRegistered() }
+        }
+
+        binding.targetButtonPcrRegisteredNoResult.apply {
+            handleColors(submissionType, SrsSubmissionType.SRS_REGISTERED_PCR)
+            setOnClickListener { viewModel.onPcrRegisteredNoResult() }
+        }
+
+        binding.targetButtonPcrNotRegistered.apply {
+            handleColors(submissionType, SrsSubmissionType.SRS_UNREGISTERED_PCR)
+            setOnClickListener { viewModel.onPcrNotRegistered() }
+        }
+
+        binding.targetButtonRapidPcr.apply {
+            handleColors(submissionType, SrsSubmissionType.SRS_RAPID_PCR)
+            setOnClickListener { viewModel.onRapidPcr() }
+        }
+
+        binding.targetButtonOther.apply {
+            handleColors(submissionType, SrsSubmissionType.SRS_OTHER)
+            setOnClickListener { viewModel.onOther() }
+        }
+
+        binding.typeSelectionNextButton.apply {
+            isEnabled = submissionType != null
+            setOnClickListener { viewModel.onNextClicked() }
+        }
+    }
+
+    private fun Button.handleColors(submissionType: SrsSubmissionType?, state: SrsSubmissionType) {
+        setTextColor(formatSrsTypeSelectionButtonTextStyleByState(context, state, submissionType))
+        backgroundTintList =
+            ColorStateList.valueOf(formatSrsTypeSelectionBackgroundButtonStyleByState(context, state, submissionType))
+    }
+
+    private fun showCancelDialog() {
+        displayDialog {
+            title(R.string.srs_cancel_dialog_title)
+            message(R.string.srs_cancel_dialog_body)
+            positiveButton(R.string.srs_cancel_dialog_positive_button)
+            negativeButton(R.string.srs_cancel_dialog_negative_button) { viewModel.onCancelConfirmed() }
+        }
+    }
+
+    companion object {
+        fun uri() = "cwa://srs-consent/openTypeSelection".toUri()
```

Comment:
you don't need a deeplink for this screen, you can open it by action from the consent screen because both them are on the same graph

Reply:
removed